tests/component/encapsulation.cpp

```diff
@@ -124,3 +125,47 @@ TEST(Encapsulation, hierarchyCircular) {
     a_parent = parent->serialise(libcellml::Format::XML);
     EXPECT_EQ(e_parent_2, a_parent);
 }
+
+TEST(Encapsulation, hierarchyWaterfallAndParse) {
+    const std::string e =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+            "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+            "<component name=\"parent_component\"/>"
+            "<component name=\"child1\"/>"
+            "<component name=\"child2\"/>"
+            "<component name=\"child3\"/>"
+            "<encapsulation>"
+            "<component_ref component=\"parent_component\">"
+            "<component_ref component=\"child1\">"
+            "<component_ref component=\"child2\">"
+            "<component_ref component=\"child3\"/>"
+            "</component_ref>"
+            "</component_ref>"
+            "</component_ref>"
+            "</encapsulation>"
+            "</model>";
```

Comment:
I think standardising indentation of XML in our tests might be better left as a general code-cleanup item after this set of changes. I've now added it to the list in #116.